jedi/inference/imports.py

```diff
@@ -245,7 +245,38 @@ def _sys_path_with_modifications(self, is_completion):
         )
 
     def follow(self):
-        if not self.import_path or not self._infer_possible:
+        if not self.import_path:
+            if self._fixed_sys_path:
+                # This is a bit of a special case, that maybe should be
```

Comment:
The description here of why the code is needed/is special feels out of step with the test which validates the behaviour. Perhaps I'm missing something, but is there more to this code than just the niche case described here?

Reply:
No it's simply that.

It's just a special case, because it breaks the basic idea of something being on the sys path.

Reply:
Ah sorry. I think if this code were here just for the special case, which this comment suggests is to do with the same name appearing at several levels in the hierarchy, then on reading this comment alone I would assume that it would only be activated by that special case and that the test to validate it would exercise that case. However the test doesn't actually seem to hit that case (the directory names it chooses differ at each level), yet this code path is taken.

Clearly the test that is present is a useful test and this codepath is useful in making it work, so perhaps this comment could be tweaked to clarify that this codepath is here for all cases of namespace packages, rather than _just_ for the case of matching names?

Reply:
I guess my comment just wasn't good enough. Because having the same names is just one case. Will try to improve :)

Reply:
I added a bit to the comment.

Once I merge this, I will probably make a release.